vector/src/main/java/im/vector/app/core/di/ActiveSessionHolder.kt

```diff
@@ -100,10 +101,16 @@ class ActiveSessionHolder @Inject constructor(
     }
 
     suspend fun getOrInitializeSession(startSync: Boolean): Session? {
-        return activeSessionReference.get() ?: sessionInitializer.tryInitialize(readCurrentSession = { activeSessionReference.get() }) { session ->
-            setActiveSession(session)
-            session.configureAndStart(applicationContext, startSyncing = startSync)
-        }
+        return activeSessionReference.get()
+                ?.also {
+                    if (startSync && !it.syncService().isSyncThreadAlive()) {
```

Comment:
for my understanding of the error scenario

- app is not started
- receive a push
- session instance is created with `startSyncing = false` and cached
- next time the is launched the sync is not started because the session instance already exists and we only `configureAndStart` brand new instances


the fix is to always start the sync (if `startStart = true`) on our cached instances if they're not already syncing?

Reply:
Yes, that's it, sorry, the code is not crystal clear.